x/debug/debug.go

```diff
@@ -0,0 +1,257 @@
+// THE SOFTWARE.
+
+package debug
+
+import (
+	"html/template"
+	"io"
+	"log"
+	"net/http"
+
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/internal/introspection"
+)
+
+var (
+	// DefaultTmpl is the default template used.
+	DefaultTmpl = template.Must(template.New("tmpl").Parse(`
+<html>
+	<head>
+	<title>/debug/yarpc</title>
+	<style type="text/css">
+		body {
+			font-family: "Courier New", Courier, monospace;
+		}
+		table {
+			color:#333333;
+			border-width: 1px;
+			border-color: #3A3A3A;
+			border-collapse: collapse;
+		}
+		table th {
+			border-width: 1px;
+			padding: 8px;
+			border-style: solid;
+			border-color: #3A3A3A;
+			background-color: #B3B3B3;
+		}
+		table td {
+			border-width: 1px;
+			padding: 8px;
+			border-style: solid;
+			border-color: #3A3A3A;
+			background-color: #ffffff;
+		}
+		header::after {
+			content: "";
+			clear: both;
+			display: table;
+		}
+		h1 {
+			width: 40%;
+			float: left;
+			margin: 0;
+		}
+		div.dependencies {
+			width: 60%;
+			float: left;
+			font-size: small;
+			text-align: right;
+		}
+	</style>
+	</head>
+	<body>
+
+<header>
+<h1>/debug/yarpc</h1>
+<div class="dependencies">
+	{{range .PackageVersions}}
+	<span>{{.Name}}={{.Version}}</span>
+	{{end}}
+</div>
+</header>
+
+{{range .Dispatchers}}
+	<hr />
+	<h2>Dispatcher "{{.Name}}" <small>({{.ID}})</small></h2>
+	<table>
+		<tr>
+			<th>Procedure</th>
+			<th>Encoding</th>
+			<th>Signature</th>
+			<th>RPC Type</th>
+		</tr>
+		{{range .Procedures}}
+		<tr>
+			<td>{{.Name}}</td>
+			<td>{{.Encoding}}</td>
+			<td>{{.Signature}}</td>
+			<td>{{.RPCType}}</td>
+		</tr>
+		{{end}}
+	</table>
+	<h3>Inbounds</h3>
+	<table>
+		<tr>
+			<th>Transport</th>
+			<th>Endpoint</th>
+			<th>State</th>
+		</tr>
+		{{range .Inbounds}}
+		<tr>
+			<td>{{.Transport}}</td>
+			<td>{{.Endpoint}}</td>
+			<td>{{.State}}</td>
+		</tr>
+		{{end}}
+	</table>
+	<h3>Outbounds</h3>
+	<table>
+		<thead>
+		<tr>
+			<th>Outbound Key</th>
+			<th>Service</th>
+			<th>Transport</th>
+			<th>RPC Type</th>
+			<th>Endpoint</th>
+			<th>State</th>
+			<th colspan="3">Chooser</th>
+		</tr>
+		<tr>
+			<th></th>
+			<th></th>
+			<th></th>
+			<th></th>
+			<th></th>
+			<th>Name</th>
+			<th>State</th>
+			<th>Peers</th>
+		</tr>
+		</thead>
+		<tbody>
+		{{range .Outbounds}}
+		<tr>
+			<td>{{.OutboundKey}}</td>
+			<td>{{.Service}}</td>
+			<td>{{.Transport}}</td>
+			<td>{{.RPCType}}</td>
+			<td>{{.Endpoint}}</td>
+			<td>{{.State}}</td>
+			<td>{{.Chooser.Name}}</td>
+			<td>{{.Chooser.State}}</td>
+			<td>
+				<ul>
+				{{range .Chooser.Peers}}
+					<li>{{.Identifier}} ({{.State}})</li>
+				{{end}}
+				</ul>
+			</td>
+		</tr>
+		</tbody>
+		{{end}}
+	</table>
+{{end}}
+	</body>
+</html>
+`))
+
+	// DefaultLogFunc is the default logging function used when there is a rendering error.
+	DefaultLogFunc = log.Printf
+)
+
+// Template represents a template created from either the html/template
+// or text/template packages.
+type Template interface {
```

Comment:
We could make this, along with `WithTemplate`, private since I really just added them for testing.